### DISCUSSION TITLE
How to install on raspberry pi is?

### DISCUSSION THREAD

**Discussion by sambarlick** [timmo001/system-bridge#2787] [Q&A]
Hi,

I am very new to raspberry pi os.

I have some limited experience with Linux using basic terminal/ssh commands and using Ubuntu and Proxmox.

I ran the install command apt install lshw upower which executed with no errors. But then the command systembridge is not found.

I have tried to research how to install a .deb file. 
I found a .deb installer but it says my Raspberry Pi 4B running 64bit OS and the .deb installer says that the file is the wrong architecture. I downloaded the .deb file of System Bridge.

What have I done wrong or missing something?

Thanks,
Sam

**Comment by aniceberg** [comment]
I was able to install the Python version on Raspian Buster by following the guide: [Headless Linux Server](https://system-bridge.timmo.dev/docs/install-python).

Afterwards, run the command `python3 -m systembridgebackend --no-gui` at the command prompt and you'll see the API-KEY.

If you want it to run persistently, created a [systemd service](https://system-bridge.timmo.dev/docs/running).

**Reply by sambarlick** [reply]
Thanks for replying.

After running the command:

python -m pip install --upgrade systembridgeshared systembridgebackend systembridgecli systembridgefrontend systembridgegui

I get the error:

error: externally-managed-environment

This environment is externally managed

To install Python packages system-wide, try apt install
python3-xyz, where xyz is the package you are trying to
install.

If you wish to install a non-Debian-packaged Python package,
create a virtual environment using python3 -m venv path/to/venv.
Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
sure you have python3-full installed.

For more information visit http://rptl.io/venv

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.

[notice] A new release of pip is available: 23.2.1 -> 23.3.1
[notice] To update, run: python -m pip install --upgrade pip

I then tried the command:

sudo apt install python3-systembridgeshared systembridgebackend systembridgecli systembridgefrontend systembridgegui

and get these errors:

E: Unable to locate package python3-systembridgeshared
E: Unable to locate package systembridgebackend
E: Unable to locate package systembridgecli
E: Unable to locate package systembridgefrontend
E: Unable to locate package systembridgegui

**Comment by aniceberg** [comment]
It looks like you're trying to install the GUI version-- the instructions I linked to above are for the CLI which I'm using since I'm running my RPi headless.

I'm not that advanced in either Linux or Python. I found that I had to run the CLI command using sudo, i.e.
`sudo python3 -m systembridgebackend --no-gui`
otherwise it won't have the required root privileges for Home Assistant's command input to control it.
